WebHostLib/api/downloads.py

```diff
@@ -0,0 +1,48 @@
+from flask import redirect, Response
+from pony.orm import commit, db_session, select
+
+from Utils import __version__
+from WebHostLib import cache
+from WebHostLib.models import ArchipelagoDownload
+from . import api_endpoints
+
+
+@cache.cached
+@api_endpoints.route("/downloads/<string:build>")
+def get_download(build: str) -> Response:
+    download_data = select(download for download in ArchipelagoDownload
+                           if download.platform == build).first()
+    if download_data and download_data.version >= __version__:
+        with db_session:
+            download_data.downloads += 1
+            commit()
+        return redirect(download_data.url)
+    return get_latest_release(build)
+
+
+def get_latest_release(build: str) -> Response:
+    """Pulls the latest release from the Archipelago GitHub and saves them to the db."""
+    import requests
+
+    response = requests.get("https://api.github.com/repos/ArchipelagoMW/Archipelago/releases/latest", timeout=5)
```

Comment:
If the timeout happens, the user get an error 404. It would be better to catch the timeout error and redirect to the last known download location, if any.
The github release json endpoint and the actual download may have different availability.